### PR TITLE
diff: save summary info in memory

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 

fix issue https://github.com/pingcap/tidb-tools/issues/362
getting the wrong summary like 
```
[2020/06/18 07:46:23.864 +00:00] [INFO] [checkpoint.go:231] ["summary info"] [instance_id=target] [schema=db] [table=txn_history] ["chunk num"=247656] ["success num"=153326] ["failed num"=0] ["ignore num"=0]
[2020/06/18 07:46:33.810 +00:00] [INFO] [checkpoint.go:231] ["summary info"] [instance_id=target] [schema=db] [table=txn_history] ["chunk num"=247656] ["success num"=153496] ["failed num"=0] ["ignore num"=0]
[2020/06/18 07:46:43.873 +00:00] [INFO] [checkpoint.go:231] ["summary info"] [instance_id=target] [schema=db] [table=txn_history] ["chunk num"=247656] ["success num"=0] ["failed num"=0] ["ignore num"=0]
```
already get 153496 success num, but get 0 success num later, don't know why now, but we can save the summary information in memory instead of select the `chunk` table every time print summary information to fix this issue,  and this will save TiDB/MySQL's performance and diff can run faster.

### What is changed and how it works?

save summary info in memory

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: pkg/diff/checkpoint.go [hunk 1]
```diff
@@ -223,21 +222,24 @@ func initTableSummary(ctx context.Context, db *sql.DB, schema, table string, con
...
 	if total == 0 {
 		// don't need to update summary info
 		return nil
 	}
 
 	log.Info("summary info", zap.String("instance_id", instanceID), zap.String("schema", schema), zap.String("table", table), zap.Int64("chunk num", total), zap.Int64("success num", successNum), zap.Int64("failed num", failedNum), zap.Int64("ignore num", ignoreNum))
 
-	state := notCheckedState
-	if total == successNum+failedNum+ignoreNum {
+	checkedNum := successNum + failedNum + ignoreNum
```
Comment (WangXiangUSTC) [inline pkg/diff/checkpoint.go:238]: minor fix

**Comment by WangXiangUSTC** [comment]
@GMHDBJD @july2993 PTAL

**Review by GMHDBJD — COMMENTED** [review]
LGTM

File: pkg/diff/diff.go [hunk 2]
```diff
@@ -450,21 +462,26 @@ func (t *TableDiff) checkChunkDataEqual(ctx context.Context, filterByRand bool,
 
 func (t *TableDiff) compareChecksum(ctx context.Context, chunk *ChunkRange) (bool, error) {
```
Comment (july2993) [inline pkg/diff/diff.go:463]: we can get the source and target checksum concurrent?

**Review by july2993 — COMMENTED** [review]
I think one reason it's slow is caused by it insert the chunk one by one, it's good to just saved in memory, but i don't think it can run faster for avoiding reading it per 10 second.
LGTM but I don't think it fixes the issue

**Comment by WangXiangUSTC** [comment]
> I think one reason it's slow is caused by it insert the chunk one by one, it's good to just saved in memory, but i don't think it can run faster for avoiding reading it per 10 second.
> LGTM but I don't think it fixes the issue

I do these refine in this PR https://github.com/pingcap/tidb-tools/pull/370, include insert chunks information in batch SQLs, and calculate source and target's checksum concurrent @july2993

File: pkg/diff/diff.go (hunk 2 above)
Comment (WangXiangUSTC) [reply pkg/diff/diff.go:463]: do it in #370

**Review by july2993 — APPROVED** [review]

**Pull request merged**